source/lac/petsc_parallel_sparse_matrix.cc

```diff
@@ -85,6 +85,28 @@ namespace PETScWrappers
       AssertThrow(ierr == 0, ExcPETScError(ierr));
     }
 
+    template <typename SparsityPatternType>
+    void
+    SparseMatrix::reinit_IS(const IndexSet &           local_rows,
+                            const IndexSet &           local_active_rows,
+                            const IndexSet &           local_columns,
+                            const IndexSet &           local_active_columns,
+                            const SparsityPatternType &sparsity_pattern,
+                            const MPI_Comm &           communicator)
+    {
+      this->communicator = communicator;
+
+      // get rid of old matrix and generate a new one
+      const PetscErrorCode ierr = destroy_matrix(matrix);
+      AssertThrow(ierr == 0, ExcPETScError(ierr));
```

Comment:
We haven't done it yet but `destroy_matrix()` should be deprecated at some point before the next release since it isn't necessary anymore (we don't support versions of PETSc which had a different destruction API), so it is better to use `MatDestroy` directly.

Reply:
Changed it as suggested